crates/automations/src/server.rs

```diff
@@ -31,9 +31,11 @@ impl Server {
         heartbeat_timeout: std::time::Duration,
         shutdown: impl std::future::Future<Output = ()>,
     ) {
+        let semaphore = Arc::new(tokio::sync::Semaphore::new(permits as usize));
```

Comment:
Let's talk through the changes to this source file -- I'm not understanding the motivation.

Reply:
Okay, I get it now. It's about being able to finely control the execution of automations within a testing context.